### ISSUE TITLE
Cannot Parse Latitude - GPX load failure

### ISSUE THREAD

**Issue by Swoddler** [cgeo/cgeo#6573]
##### Detailed steps causing the problem:
- Load caches from GPX file in Dropbox

##### Actual behavior after performing these steps:
- Get a message saying "Import failed - unexpected error   Cannot Parse Latitiude"
- this consistently happens after 1678 caches (of the 4500 in the gpx)

##### Expected behavior after performing these steps:
- I have used this method for years and never had a problem.

##### Version of c:geo used:
- 2017.05.09

##### Is the problem reproducible for you?
Yes

##### System information:
```
Attach system information here if available (see c:geo menu -> About c:geo -> Swipe right to System)
Keep the apostrophe at beginning and end to have it properly formatted
```

##### Other comments and remarks:

- I have rebuilt the GSAK database but it hasn't helped

**Comment by UniQP** [comment]
Can you provide the GPX file?

**Comment by Swoddler** [comment]
It is too big to email, but I have hopefully sent you a Dropbox link. Let
me know if I didn't please
Mark

**Comment by Swoddler** [comment]
https://www.dropbox.com/s/tcpo872lejn689k/Home%204500.gpx?dl=0

**Comment by UniQP** [comment]
I can imagine that `<wpt lat="56" lon="-5">` of http://coord.info/GCNKHZ causes the problem in combination with https://github.com/cgeo/cgeo/pull/5790.

If so, [GCNKHZ.gpx.zip](https://github.com/cgeo/cgeo/files/1016716/GCNKHZ.gpx.zip) provides a reduced test.

**Comment by Swoddler** [comment]
That''s great. I shall check this out later on today

On 21 May 2017 8:27 a.m., "Sebastian Buchwald" <notifications@github.com>
wrote:

I can imagine that <wpt lat="56" lon="-5"> of http://coord.info/GCNKHZ
causes the problem in combination with #5790
<https://github.com/cgeo/cgeo/pull/5790>.

If so, GCNKHZ.gpx.zip
<https://github.com/cgeo/cgeo/files/1016716/GCNKHZ.gpx.zip> provides a
reduced test.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/cgeo/cgeo/issues/6573#issuecomment-302920250>, or mute
the thread

**Comment by UniQP** [comment]
I my reduced test it worked, so investigated a bit further. The problem is more subtle: You provided corrected coordinates for http://coord.info/GC5WZFM and we fail to parse `<gsak:LatBeforeCorrect>56</gsak:LatBeforeCorrect>`.

I will provide a PR with a fix.

**Comment by Swoddler** [comment]
Yes, I cannot get this cache to load and it is preventing my whole database from loading. The coordinates were corrected on the Groundspeak site, not in GSAK. I need a fix for this please. The GPX will load into GSAK and GDAK. It's just cgeo that cannot load it......

**Comment by UniQP** [comment]
The issue was fixed in e4cc7f02c245bb2f9317c4cf0650c4532995d11f, which will be contained in the next release.

As a workaround, you can manually change `56` to `56.0` in the above mentioned line, or remove the corrected coordinates from http://coord.info/GC5WZFM (you can add them to the personal cache note instead).

I also opened #6575 as a follow-up issue.

**Comment by Swoddler** [comment]
Thanks Sebastian. You've been a big help!
Kind regards
Mark
